# IMPORT LIBRARIES


In [68]:
import pandas as pd
import numpy as np

## 1.1 Basic information

In [69]:
df = pd.read_csv('../data/creditcard.csv')
print("=== BASIC INFORMATION ===")
print(f"Data have {df.shape[0]} rows and {df.shape[1]} columns")
print(f"\nColumns name: \n{df.columns}")
print(f"\nDatatype of column \n{df.dtypes}")
print(f"\nFive first rows: \n{df.head().round(2)}")
print(f"\nDescriptive statistics \n{df.describe().round(2)}")

=== BASIC INFORMATION ===
Data have 284807 rows and 31 columns

Columns name: 
Index(['Time', 'V1', 'V2', 'V3', 'V4', 'V5', 'V6', 'V7', 'V8', 'V9', 'V10',
       'V11', 'V12', 'V13', 'V14', 'V15', 'V16', 'V17', 'V18', 'V19', 'V20',
       'V21', 'V22', 'V23', 'V24', 'V25', 'V26', 'V27', 'V28', 'Amount',
       'Class'],
      dtype='object')

Datatype of column 
Time      float64
V1        float64
V2        float64
V3        float64
V4        float64
V5        float64
V6        float64
V7        float64
V8        float64
V9        float64
V10       float64
V11       float64
V12       float64
V13       float64
V14       float64
V15       float64
V16       float64
V17       float64
V18       float64
V19       float64
V20       float64
V21       float64
V22       float64
V23       float64
V24       float64
V25       float64
V26       float64
V27       float64
V28       float64
Amount    float64
Class       int64
dtype: object

Five first rows: 
   Time    V1    V2    V3    V4    V5    V6 

# Data Overview: Credit Card Fraud Detection

---

### 1. Dataset Dimensions
<div style="background-color: #1c6fc2; padding: 15px; border-left: 5px solid #343a40;">
    <ul>
        <li><b>Total Observations:</b> 284,807 rows</li>
        <li><b>Total Features:</b> 31 columns</li>
        <li><b>Missing Values:</b> 0 (The dataset is complete with no null entries)</li>
    </ul>
</div>

### 2. Feature Composition

| Feature Group | Count | Data Type | Description |
| :--- | :---: | :---: | :--- |
| **PCA Components** (`V1` - `V28`) | 28 | `float64` | Transformed numerical features for privacy/anonymization. |
| **Metadata** (`Time`, `Amount`) | 2 | `float64` | Seconds elapsed since the first transaction and the transaction value. |
| **Target** (`Class`) | 1 | `int64` | Binary label (0: Legitimate, 1: Fraudulent). |

### 3. Key Statistical Observations

> **Initial Analysis:**

* **High Variance in Amount:** The `Amount` column shows a mean of $88.35$ but a standard deviation of $250.12$. The maximum value reaches $25,691.16$, indicating significant outliers.
* **Severe Class Imbalance:** The mean of the `Class` column is

## 1.2 Check missing values

In [70]:
missing_data = df.isnull().sum()
missing_percentage = missing_data / len(df) * 100
print(missing_data)
print(missing_percentage)

Time      0
V1        0
V2        0
V3        0
V4        0
V5        0
V6        0
V7        0
V8        0
V9        0
V10       0
V11       0
V12       0
V13       0
V14       0
V15       0
V16       0
V17       0
V18       0
V19       0
V20       0
V21       0
V22       0
V23       0
V24       0
V25       0
V26       0
V27       0
V28       0
Amount    0
Class     0
dtype: int64
Time      0.0
V1        0.0
V2        0.0
V3        0.0
V4        0.0
V5        0.0
V6        0.0
V7        0.0
V8        0.0
V9        0.0
V10       0.0
V11       0.0
V12       0.0
V13       0.0
V14       0.0
V15       0.0
V16       0.0
V17       0.0
V18       0.0
V19       0.0
V20       0.0
V21       0.0
V22       0.0
V23       0.0
V24       0.0
V25       0.0
V26       0.0
V27       0.0
V28       0.0
Amount    0.0
Class     0.0
dtype: float64


**There is no missing value in dataset**

## 1.3 Check Duplicate value

In [71]:
duplicated_value = df.duplicated().sum()
duplicated_percentage = duplicated_value / len(df) * 100
print(f"Dataset have {duplicated_value} duplicated rows with {duplicated_percentage:.1f} percent")

Dataset have 1081 duplicated rows with 0.4 percent


### 4. Duplicate Value Treatment

A total of **1,081 duplicate rows** (~0.4%) were identified. These records were removed to ensure data quality and model reliability based on the following:

* **Statistical Impossibility:** Given the 15-decimal precision of PCA features ($V1$-$V28$), identical rows are likely logging errors rather than unique transactions.
* **Preventing Data Leakage:** Removing duplicates prevents the same observation from appearing in both training and testing sets, which would lead to over-optimistic (fake) performance metrics.
* **Model Generalization:** Eliminating redundant data points prevents the model from over-fitting to specific identical samples, ensuring it learns general patterns of fraud.

**Action taken:** `df.drop_duplicates(inplace=True)`

In [72]:
df = df.drop_duplicates()
final_duplicates = df.duplicated().sum()

In [73]:
print(df.duplicated().sum())

0
